fre/yamltools/combine_yamls.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env python
+"""
+    Script combines the model yaml with 
+    the compile, platform, and experiment
+    yamls.
+"""
+
+## TO-DO:
+# - figure out way to safe_load (yaml_loader=yaml.SafeLoader?)
+# - condition where there are multiple pp and analysis yamls
+# - fix schema for validation
+
+import os
+import json
+import shutil
+from pathlib import Path
+import click
+from jsonschema import validate
+import yaml
+
+def join_constructor(loader, node):
+    """
+    Allows FRE properties defined
+    in main yaml to be concatenated.
+    """
+    seq = loader.construct_sequence(node)
+    return ''.join([str(i) for i in seq])
+
+def yaml_load(yamlfile):
+    """
+    Load the yamlfile
+    """
+    with open(yamlfile, 'r') as yf:
+        y = yaml.load(yf,Loader=yaml.Loader)
+
+    return y
+
+def combine_model(modelyaml,combined,experiment,platform,target):
+    """
+    Create the combined.yaml and merge it with the model yaml
+    Arguments:
+        modelyaml  :  model yaml file
+        combined   :  final combined file name
+        experiment :  experiment name
+        platform   :  platform used
+        target     :  targets used
+    """
+    # copy model yaml info into combined yaml
+    with open(combined,'w+',encoding='UTF-8') as f1:
+        f1.write(f'name: &name "{experiment}"\n')
+        f1.write(f'platform: &platform "{platform}"\n')
+        f1.write(f'target: &target "{target}"\n\n')
+        with open(modelyaml,'r',encoding='UTF-8') as f2:
+            f1.write("### MODEL YAML SETTINGS ###\n")
+            shutil.copyfileobj(f2,f1)
+    print(f"   model yaml: {modelyaml}")
+
+def experiment_check(mainyaml_dir,comb,experiment):
+    """
+    Check that the experiment given is an experiment listed in the model yaml.
+    Extract experiment specific information and file paths.
+    Arguments:
+        mainyaml_dir  :  model yaml file
+        comb            :  combined yaml file name
+        experiment      :  experiment name
+    """
+    comb_model=yaml_load(comb)
+
+    # Check if exp name given is actually valid experiment listed in combined yaml
+    exp_list = []
+    for i in comb_model.get("experiments"):
+        exp_list.append(i.get("name"))
+
+    if experiment not in exp_list:
+        raise NameError(f"{experiment} is not in the list of experiments")
+
+    # set platform yaml filepath
+    if comb_model["shared"]["compile"]["platformYaml"] is not None:
+        py=comb_model["shared"]["compile"]["platformYaml"]
+        py_path=Path(os.path.join(mainyaml_dir,py))
+    else:
+        py=False
```

Comment:
seems unused?

Reply:
meant to change `py=False` to `py_path=False`